float64[] reference
float64[] input_ref
float64[] states
float64[] disturbance
---
float64[] control_signals
